### DISCUSSION TITLE
Import Uno Style-Reset in a CSS Layer

### DISCUSSION THREAD

**Discussion by jamesarosen** [unocss/unocss#4232] [Q&A]
[unocss/unocss#3384](https://github.com/unocss/unocss/issues/3384) shows how to import the UnoCSS layers as CSS layers using the PostCSS plugin. That works great!

I'd like to put one of the [UnoCSS style-reset](https://unocss.dev/guide/style-reset) sheets into a CSS layer as well. Does anyone have a good solution for that?

Option 1: copy the CSS and wrap it in a `@layer base.reset {...}`. This works, but it increases maintenance costs.

Option 2: if using Vite, import `import whatever 'from @unocss/reset/whatever.css?inline'`, then wrap the result in `@layer base.reset {...}`. This works for Vite, but not for other systems.

**Comment by jamesarosen** [comment]
Another option would be to expose the resets as UnoCSS preflights and give them _Uno_ layers rather than CSS `@layer`s. Application developers could then control where they end up with `outputToCssLayers`.

**Reply by jamesarosen** [reply]
Ah! This actually suggests a workaround:

```ts
import fs from 'node:fs/promises'

// uno.config.ts
export default defineConfig({
  preflights: [
    {
      layer: 'base.reset',
      getCSS: async () => {
        return fs.readFile('node_modules/@unocss/reset/tailwind-compat.css', { encoding: 'utf-8' })
      },
    },
  ],
})
```

That puts the `tailwind-compat.css` contents into the `base.reset` CSS `@layer`.

**Comment by henrikvilhelmberglund** [comment]
Another solution:

1. Create `reset-in-layer.css`
2. In that file `@import "@unocss/reset/tailwind.css" layer(preflights);` (or whatever reset you like)
3. Import `reset-in-layer.css` instead of the reset in your entry point

**Reply by jamesarosen** [reply]
This doesn't seem to work with Webpack's CSS loader. It results in
```html
<style media="layer(preflights)">
/* the correct CSS */
</style>
```
`layer(preflights)` isn't a valid media query. This is probably related to [this Webpack bug](https://github.com/webpack-contrib/css-loader/issues/1532).

**Reply by henrikvilhelmberglund** [reply]
You need

```ts
    experiments: {
      css: true,
    },
```

eg. webpack 5 (or 6)
